respect-credentials/src/commonMain/kotlin/world/respect/credentials/passkey/username/CreateCredentialUsernameUseCase.kt

```diff
@@ -0,0 +1,46 @@
+package world.respect.credentials.passkey.username
+
+import io.ktor.http.Url
+import io.ktor.http.authority
+import io.ktor.http.isSecure
+
+/**
+ * When saving credentials we can only save a 'username', we also need to know the learning space
+ * url. Linking each credential strictly to a learning space origin would make it impossible to show
+ * a simple dialog when the user first starts the app to select an account from any learning space.
+ *
+ * The credential username is used both as user.name for a passkey and the username when saving a
+ * username and password to a password manager.
+ *
+ * The username is visible to the user so we want this to be as user-friendly as possible.
+ * The credential username will be as follows:
+ *
+ * username@example.org where the LearningSpace uses https and the standard port
+ * username@example.org:port where the LearningSpace uses https and a non-standard port
+ * username@http://example.org/ where the LearningSpace does not use https
+ *
+ */
+class CreateCredentialUsernameUseCase(
```

Comment:
Credentialusername Create/parse should be removed from respect - not needed because the url is known at the time of logging in / requesting or creating credentials.